airbyte-integrations/bases/standard-destination-test/src/main/java/io/airbyte/integrations/standardtest/destination/DestinationAcceptanceTest.java

```diff
@@ -376,7 +377,15 @@ public void testSyncWithLargeRecordBatch(final String messagesFilename, final St
     final List<AirbyteMessage> messages = MoreResources.readResource(messagesFilename).lines()
         .map(record -> Jsons.deserialize(record, AirbyteMessage.class)).collect(Collectors.toList());
 
-    final List<AirbyteMessage> largeNumberRecords = Collections.nCopies(400, messages).stream().flatMap(List::stream).collect(Collectors.toList());
+    final List<AirbyteMessage> largeNumberRecords = Collections
+        .nCopies(400, messages)
+        .stream()
+        .flatMap(List::stream)
+        // regroup messages per stream
+        .sorted(Comparator
+            .comparing(AirbyteMessage::getType)
+            .thenComparing(message -> message.getType().equals(Type.RECORD) ? message.getRecord().getStream() : message.toString()))
```

Comment:
This integration tests is generating "large" number of records by cloning 400 copies of the "test" message resource file.

But this means that streams are interleaved and this is killing the performance of the buffering (limited by the number of concurrent buffers at once).

So I have to sort the list of messages per stream so they are batched together instead.

I will create a new issue to record this limitation if we need to address this in the future instead...